src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -329,7 +333,14 @@ public static string GetDefaultAddress(int pid)
 
             if (!TryGetDefaultAddress(pid, out string defaultAddress))
             {
-                throw new ServerNotAvailableException($"Process {pid} not running compatible .NET runtime.");
+                string msg = $"Process {pid} not running compatible .NET runtime.";
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    msg += $" Please verify that {IpcRootPath} is writable by the current user. "
```

Comment:
If this is the advice for this failure, you could simply check the if the TMPDIR environment variable is set and base the recommendation on that.  Either through `Environment.GetEnvironmentVariable("TMPDIR");` or however it decides where to look?  IE if "tmpdir" is set recommend that with the path to check, and if it's not, then it's a different issue?

Reply:
I'm not sure if we can reliably check the target process's $TMPDIR environment variable, so I am assuming this is something the user needs to do.  https://aka.ms/dotnet-diagnostics-port  describes the issue.

Reply:
Yeah, we can check our local process $TMPDIR but we can't check the target process's $TMPDIR unless we have root permission (as far as I know).

Reply:
I think the underlying problem is that we have a protocol that depends on a process TMPDIR to begin with, if they are not the same, we won't find default IPC channels based on pid. This recently became even more problematic on sandboxed MacOS/MacCatalyst apps, since their TMPDIR is rooted into their sandboxed file system, meaning our dotnet-* diagnostic tools will never be able to detect any sandboxed apps IPC channels. There are more problems with sandboxed IPC channels than just TMPDIR, see https://github.com/dotnet/runtime/issues/94114#issuecomment-1784761813. The sandboxing issues might call for alternative solutions in the end, like using TCP/IP and dotnet-dsrouter, but for the default IPC channel, maybe we should have a more stable root, in current users home folder?

Reply:
Changing the default Diagnostic IPC location to a stable location will require a runtime change.  Feel free to open an issue requesting the change.